cmd/ops_agent_uap_plugin/service_linux.go

```diff
@@ -0,0 +1,201 @@
+func (ps *OpsAgentPluginServer) GetStatus(ctx context.Context, msg *pb.GetStatusRequest) (*pb.Status, error) {
+	log.Println("Received a GetStatus request")
+	if ps.cancel == nil {
+		log.Println("The Ops Agent plugin is not running")
+		return &pb.Status{Code: 1, Results: []string{"The Ops Agent Plugin is not running."}}, nil
+	}
+	log.Println("The Ops Agent plugin is running")
+	return &pb.Status{Code: 0, Results: []string{"The Ops Agent Plugin is running ok."}}, nil
+}
+
+func runCommand(cmd *exec.Cmd) (string, error) {
+	if cmd == nil {
+		return "", nil
+	}
+	cmd.SysProcAttr = &syscall.SysProcAttr{
+		Pdeathsig: syscall.SIGKILL,
+	}
+	log.Printf("Running command: %s", cmd.Args)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		log.Printf("Command %s failed, \ncommand output: %s\ncommand error: %s", cmd.Args, string(out), err)
+		return string(out), err
+	}
+	return string(out), nil
+}
+
+func validateOpsAgentConfig(ctx context.Context, runCommand RunCommandFunc, pluginBaseLocation string) error {
+	configValidationCmd := exec.CommandContext(ctx,
+		pluginBaseLocation+"/"+ConfGeneratorBinary,
+		"-in", OpsAgentConfigLocationLinux,
+	)
+	if output, err := runCommand(configValidationCmd); err != nil {
+		return fmt.Errorf("failed to validate the Ops Agent config:\ncommand output: %s\ncommand error: %s", output, err)
+	}
+	return nil
+}
+
+func generateSubagentConfigs(ctx context.Context, runCommand RunCommandFunc, pluginBaseLocation string) error {
+	confGeneratorBinaryFullPath := fmt.Sprintf("%s/%s", pluginBaseLocation, ConfGeneratorBinary)
+	otelConfigGenerationCmd := exec.CommandContext(ctx,
+		confGeneratorBinaryFullPath,
+		"-service", "otel",
+		"-in", OpsAgentConfigLocationLinux,
+		"-out", pluginBaseLocation+"/"+OtelRuntimeDirectory,
+		"-logs", pluginBaseLocation+"/"+LogsDirectory)
+
+	if output, err := runCommand(otelConfigGenerationCmd); err != nil {
+		return fmt.Errorf("failed to generate Otel config:\ncommand output: %s\ncommand error: %s", output, err)
+	}
+
+	fluentBitConfigGenerationCmd := exec.CommandContext(ctx,
+		confGeneratorBinaryFullPath,
+		"-service", "fluentbit",
+		"-in", OpsAgentConfigLocationLinux,
+		"-out", pluginBaseLocation+"/"+FluentBitRuntimeDirectory,
+		"-logs", pluginBaseLocation+"/"+LogsDirectory, "-state", pluginBaseLocation+"/"+FluentBitStateDiectory)
+
+	if output, err := runCommand(fluentBitConfigGenerationCmd); err != nil {
+		return fmt.Errorf("failed to generate Fluntbit config:\ncommand output: %s\ncommand error: %s", output, err)
+	}
+	return nil
+}
+
+func findPreExistentAgents(ctx context.Context, runCommand RunCommandFunc, agentSystemdServiceNames []string) (bool, error) {
+	cmdArgs := []string{"systemctl", "list-unit-files"}
+	cmdArgs = append(cmdArgs, agentSystemdServiceNames...)
+	findOpsAgentCmd := exec.CommandContext(ctx,
+		cmdArgs[0], cmdArgs[1:]...,
+	)
+	output, err := runCommand(findOpsAgentCmd)
+	if strings.Contains(output, "0 unit files listed.") {
```

Comment:
What happens if the output return "10 unit files listed."?

Reply:
If the output returns `non-zero` unit files listed, it implies that there are existing agent installations and `Start()` method terminates with errors.